docs/source/usage/performance-tuning.rst

```diff
@@ -0,0 +1,194 @@
+        OR (
+            (L.h3_res = 9 AND h3_toparent(R.h3_15, 9) = L.h3_cell) -- RES=9 MATCH?
+        AND(
+            L.h3_is_core OR st_contains(L.h3_wkb, R.geom_wkb)))
+
+Adjust the query as follows for results that strip back the H3 information (more akin to traditional ``ST_`` functions):
+
+.. code-block:: sql
+
+    -- RESULTS WITHOUT H3 INFORMATION @ GEOMETRY-CENTRIC
+    SELECT DISTINCT * except(L.h3_res, L.h3_is_core, L.h3_cell, L.h3_wkb, R.h3_15, R.geom_wkb)
+    FROM poly_tbl as L, pnt_tbl as R
+    WHERE 
+        (L.h3_res = 5 AND h3_toparent(R.h3_15, 5) = L.h3_cell)   -- CORE RES=5 MATCH?
+        OR (
+            (L.h3_res = 9 AND h3_toparent(R.h3_15, 9) = L.h3_cell) -- RES=9 MATCH?
+        AND(
+            L.h3_is_core OR st_contains(L.h3_wkb, R.geom_wkb)))
+                                                                                                                                                                                                             
+Databricks Photon APIs
+######################                                                                                                                                                                                                             
+
+Since DBR 11.2, Databricks `Photon Runtimes <https://docs.databricks.com/runtime/photon.html>`__ have delivered `H3 APIs <https://docs.databricks.com/sql/language-manual/sql-ref-h3-geospatial-functions.html>`__ that are blazing fast. Mosaic leverages these APIs everywhere possible in its implementation. They are pre-loaded in SQL, meaning Spark SQL on DBR as well as DBSQL. 
+
+For python: 
+
+.. code-block:: py
+
+    from pyspark.databricks.sql.functions import *
+
+For scala: 
+
+.. code-block:: scala
+
+    import com.databricks.sql.functions._
+
+Tuning Tips
+###########                                             
+
+Spark `Adaptive Query Exection (AQE) <https://spark.apache.org/docs/latest/sql-performance-tuning.html#adaptive-query-execution>`__ is tuned for data-heavy processing where each row has a cost. It likes those rows to be even and has a hard time reasoning about hidden compute costs as incurred when ``ST_`` functions, e.g. ``ST_Contains`` are invoked. To bring more control AQE can be turned completely off:
+
+.. code-block:: py
+
+    spark.conf.set("spark.databricks.optimizer.adaptive.enabled", False) # <- default is True
+    spark.conf.set("spark.sql.adaptive.enabled", False)                  # <- default is True
+
+AQE can also be left on with other controls more finely managing its behavior, such as:
+
+.. code-block:: py
+
+    spark.conf.set("spark.sql.adaptive.coalescePartitions.enabled", True)                  # <- default is True, may want False
+    spark.conf.set("spark.sql.adaptive.coalescePartitions.parallelismFirst", False)        # <- default is True (respect size)
+    spark.conf.set("spark.sql.adaptive.advisoryPartitionSizeInBytes", "24MB")              # <- default is 64MB
```

Comment:
I would go even lower here, but it is very hard to say what is the correct number

Reply:
I have messed with others (16MB) and frankly have landed on 24MB as a good goldilocks zone that is ok for smaller and larger data.